brokerapi/brokers/gcp_service_broker.go

```diff
@@ -396,18 +396,30 @@ func (gcpBroker *GCPServiceBroker) LastOperation(instanceID string) (models.Last
 		return models.LastOperation{}, models.ErrInstanceDoesNotExist
 	}
 
-	if gcpBroker.ServiceBrokerMap[instance.ServiceId].Async() {
+	if gcpBroker.ServiceBrokerMap[instance.ServiceId].ProvisionsAsync() || gcpBroker.ServiceBrokerMap[instance.ServiceId].DeprovisionsAsync() {
```

Comment:
these branches are getting pretty deep. Recommend keeping the branch but pullung them out to their own local/private functions

eg:

```go
if async {
   return gcpBroker.lastOperationAsync()
} else {
   return gcpBroker.lastOperationSync()
}
```